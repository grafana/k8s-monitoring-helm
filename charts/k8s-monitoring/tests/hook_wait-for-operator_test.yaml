---
suite: Test operator readiness hook functionality
templates:
  - templates/hooks/post-install_wait-for-operator.yaml
tests:
  - it: should create hook when operator is deployed and readiness is enabled
    set:
      alloy-operator:
        deploy: true
        waitForReadiness:
          enabled: true
      alloy-metrics:
        enabled: true
        extraConfig: ""
    asserts:
      - hasDocuments:
          count: 4
      - documentIndex: 0
        matchSnapshot:
          path: spec

  - it: should not create hook when operator is not deployed
    set:
      alloy-operator:
        deploy: false
        waitForReadiness:
          enabled: true
      clusterMetrics:
        enabled: true
      destinations:
        - name: prometheus
          type: prometheus
    asserts:
      - hasDocuments:
          count: 0

  - it: should not create hook when readiness wait is disabled
    set:
      alloy-operator:
        deploy: true
        waitForReadiness:
          enabled: false
      clusterMetrics:
        enabled: true
      destinations:
        - name: prometheus
          type: prometheus
    asserts:
      - hasDocuments:
          count: 0
