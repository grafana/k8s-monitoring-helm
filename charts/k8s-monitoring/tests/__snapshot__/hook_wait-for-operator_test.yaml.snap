should create hook when operator is deployed and readiness is enabled:
  1: |
    backoffLimit: 3
    template:
      metadata:
        labels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: release-name-k8s-monitoring
          sidecar.istio.io/inject: "false"
        name: release-name-k8s-monitoring-wait-alloy-operator
      spec:
        containers:
          - command:
              - /bin/bash
              - -ce
              - |
                echo "Waiting for Alloy Operator to be ready..."
                kubectl wait \
                  --timeout=300s \
                  --namespace=NAMESPACE \
                  --for=condition=available \
                  deployment/RELEASE-NAME-alloy-operator

                echo "Verifying Alloy Operator pod is running..."
                kubectl wait \
                  --timeout=120s \
                  --namespace=NAMESPACE \
                  --for=condition=ready \
                  pod -l app.kubernetes.io/name=alloy-operator

                echo "Alloy Operator is ready!"
            image: ghcr.io/grafana/helm-chart-toolbox-kubectl:0.1.1
            imagePullPolicy: IfNotPresent
            name: wait-for-operator
            securityContext:
              allowPrivilegeEscalation: false
              capabilities:
                drop:
                  - ALL
              readOnlyRootFilesystem: true
              seccompProfile:
                type: RuntimeDefault
        nodeSelector:
          kubernetes.io/os: linux
        restartPolicy: Never
        serviceAccountName: release-name-k8s-monitoring-wait-alloy-operator
    ttlSecondsAfterFinished: 300
